src/Config/ConfigResolver.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Slim\Console\Config;
+
+use InvalidArgumentException;
+use ReflectionClass;
+use Slim\Console\Exception\ConfigNotFoundException;
+
+use function array_filter;
+use function file_exists;
+use function file_get_contents;
+use function implode;
+use function is_array;
+use function is_null;
+use function json_decode;
+use function strpos;
+
+class ConfigResolver
+{
+    public const FORMAT_JSON = 'json';
+    public const FORMAT_PHP = 'php';
+    //public const FORMAT_YAML = 'yaml';
+    private const CONFIG_FILENAME = 'slim-console.config';
+
+    /** @var string|null */
+    private $file = null;
+
+    /** @var string */
+    private $dir;
+
+    public function __construct(string $dir)
+    {
+        $this->dir = $dir;
+    }
+
+    public function loadConfig(): ?Config
+    {
+        $this->locateConfig();
+
+        return $this->loadConfigByFormat();
+    }
+
+    public function getFile(): string
+    {
+        return $this->file ?? "";
+    }
+
+    protected function locateConfig(): void
+    {
+        $reflection = new ReflectionClass($this);
+        $filterFormats = function ($value, $key) {
+            return strpos($key, 'FORMAT') === 0;
+        };
+        $formats = array_filter($reflection->getConstants(), $filterFormats, ARRAY_FILTER_USE_BOTH);
```

Comment:
While this is clever, we should instead make use of a regular array that contains the supported formats instead. It'll avoid having to use costly `ReflectionClass` and unnecessary parsing logic.

Let's do this instead, this way the `Config` object can be extended and overriden:
```php
protected $supportedFormats = [
    self::FORMAT_PHP,
    self::FORMAT_JSON,
];